src/EditorFeatures/Core/Implementation/CodeCleanup/CodeCleanupService.cs

```diff
@@ -0,0 +1,165 @@
+
+        private readonly ICodeFixService _codeFixService;
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public CodeCleanupService(
+            ICodeFixService codeFixService)
+        {
+            _codeFixService = codeFixService;
+        }
+        public static IDictionary<PerLanguageOption<bool>, string[]> Dictionary
+        {
+            get
+            {
+                return _dictionary;
+            }
+        }
+
+        private static ImmutableDictionary<PerLanguageOption<bool>, string[]> GetCodeCleanupOptionMapping()
+        {
+            var dictionary = new Dictionary<PerLanguageOption<bool>, string[]>();
+            dictionary.Add(FeatureOnOffOptions.FixImplicitExplicitType,
+                new[] { IDEDiagnosticIds.UseImplicitTypeDiagnosticId, IDEDiagnosticIds.UseExplicitTypeDiagnosticId });
+            dictionary.Add(FeatureOnOffOptions.FixThisQualification,
+                new[] { IDEDiagnosticIds.AddQualificationDiagnosticId, IDEDiagnosticIds.RemoveQualificationDiagnosticId });
+            dictionary.Add(FeatureOnOffOptions.FixFrameworkTypes,
+                new[] { IDEDiagnosticIds.PreferFrameworkTypeInDeclarationsDiagnosticId, IDEDiagnosticIds.PreferFrameworkTypeInMemberAccessDiagnosticId });
+            dictionary.Add(FeatureOnOffOptions.FixAddRemoveBraces,
+                new[] { IDEDiagnosticIds.AddBracesDiagnosticId });
+            dictionary.Add(FeatureOnOffOptions.FixAccessibilityModifiers,
+                new[] { IDEDiagnosticIds.AddAccessibilityModifiersDiagnosticId });
+            dictionary.Add(FeatureOnOffOptions.SortAccessibilityModifiers,
+                new[] { IDEDiagnosticIds.OrderModifiersDiagnosticId });
+            dictionary.Add(FeatureOnOffOptions.MakeReadonly,
+                new[] { IDEDiagnosticIds.MakeFieldReadonlyDiagnosticId });
+            dictionary.Add(FeatureOnOffOptions.RemoveUnnecessaryCasts,
+                new[] { IDEDiagnosticIds.RemoveUnnecessaryCastDiagnosticId });
+            dictionary.Add(FeatureOnOffOptions.FixExpressionBodiedMembers,
+                new[] { IDEDiagnosticIds.UseExpressionBodyForConstructorsDiagnosticId,
+                IDEDiagnosticIds.UseExpressionBodyForMethodsDiagnosticId,
+                IDEDiagnosticIds.UseExpressionBodyForConversionOperatorsDiagnosticId,
+                IDEDiagnosticIds.UseExpressionBodyForOperatorsDiagnosticId,
+                IDEDiagnosticIds.UseExpressionBodyForPropertiesDiagnosticId,
+                IDEDiagnosticIds.UseExpressionBodyForIndexersDiagnosticId,
+                IDEDiagnosticIds.UseExpressionBodyForAccessorsDiagnosticId });
+            dictionary.Add(FeatureOnOffOptions.FixInlineVariableDeclarations,
+                new[] { IDEDiagnosticIds.InlineDeclarationDiagnosticId });
+            dictionary.Add(FeatureOnOffOptions.RemoveUnusedVariables,
+                new[] { "CS0168", "CS0219" });
```

Comment:
this seems super brittle.

Reply:
added them to the IDEDiagnosticIds... Not sure if there is a better place for them

Reply:
I'm trying to make two points here:
1. it doesn't make sense htat C# specific diagnostic IDs are listed in this 'common' layer.
2. it seems brittle to me that this feature has to know about all these diagnostic IDs.  That's a lot of duplication.  Instead, this feature shouldn't know the diagnostic IDs, but should instead 'discover' the, through some mechanism.  That would allow hte *other* features that want to be involved in code-cleanup to somehow 'declare' that they shuld be involved, and in such a way that thsi information can be discovered here.

Reply:
@sharwell  could you help with this design? Thanks